src/main/java/uk/gov/justice/hmpps/prison/util/SQLProvider.java

```diff
@@ -1,153 +0,0 @@
-import org.springframework.stereotype.Component;
-import uk.gov.justice.hmpps.prison.exception.PrisonApiRuntimeException;
-
-import java.io.BufferedReader;
-import java.io.CharArrayWriter;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.text.ParseException;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
-
-@Component
-public class SQLProvider {
-    private final Logger log = LoggerFactory.getLogger(getClass());
-
-    private final Map<String, String> statements = new HashMap<>();
-
-    private final ApplicationContext applicationContext;
-    private final String schemaType;
-
-    @Autowired
-    public SQLProvider(final ApplicationContext applicationContext, @Value("${schema.type}") final String schemaType) {
-        this.applicationContext = applicationContext;
-        this.schemaType = schemaType;
-
-    }
-
-    public void loadSql(final String className) {
-        loadSqlResource("classpath:sqls/" + className + ".sql");
-        final var schemas = StringUtils.split(schemaType, ",");
-        if (schemas != null) {
-            Arrays.asList(schemas).forEach(schema -> loadSqlResource("classpath:sqls/" + schema + "/" + className + ".sql"));
-        }
-    }
-
-    private void loadSqlResource(final String resourcePath) {
-        final var resource = applicationContext.getResource(resourcePath);
-        if (resource.exists()) {
-            try {
-                log.debug("Loading resource {}", resourcePath);
-                loadFromStream(resource.getInputStream());
-            } catch (final IOException ex) {
-                log.error(ex.getMessage(), ex);
-            }
-        }
-    }
-
-    private void loadFromStream(final InputStream in) {
-        final var out = new CharArrayWriter();
-        final var cbuf = new char[1024];
-        try {
-            final var reader = new BufferedReader(new InputStreamReader(in));
-            var size = reader.read(cbuf);
-            while (size > -1) {
-                out.write(cbuf, 0, size);
-                size = reader.read(cbuf);
-            }
-            parse(out.toCharArray());
-        } catch (final Exception ex) {
-            throw new PrisonApiRuntimeException(ex);
-        }
-    }
-
-    private int getNext(final char[] content, final int offset, final char searchFor) {
-        if (offset >= 0) {
-            for (var i = offset; i < content.length; i++) {
-                if (content[i] == searchFor) return i;
-            }
-        }
-        return -1;
-    }
-
-
-    private String makeString(final char[] content, final int startIndex, final int endIndex) {
-        final var sb = new StringBuilder();
-        for (var i = startIndex; i < content.length && i < endIndex; i++) {
-            sb.append(content[i]);
-        }
-        return sb.toString();
-    }
-
-    private void parse(final char[] content) throws ParseException {
```

Comment:
i noticed that the old sql files get run through this when loaded

can we be confident that the copied queries are the same, now they are being read directly?

Reply:
@andymarke @Mjwillis any thoughts on this?